src/Kernel/Class.class.st

```diff
@@ -870,21 +870,23 @@ Class >> removeFromSystem: logged [
 	"Forget the receiver from the Smalltalk global dictionary. Any existing  instances will refer to an obsolete version of the receiver.
 	Keep the class name and category for triggering the system change message. If we wait to long, then we get obsolete information which is not what we want.
 	Tell class to deactivate and unload itself-- two separate events in the module system"
+
 	| myCategory |
 	self release.
 	self unload.
 
-	self superclass ifNotNil:
-		["If we have no superclass there's nothing to be remembered"
-		self superclass addObsoleteSubclass: self].
+	self superclass ifNotNil: [ "If we have no superclass there's nothing to be remembered" self superclass addObsoleteSubclass: self ].
 
 	"we add the class to Undeclared so that if references still exist, they will  be automatically fixed
 	if this class is loaded again. We do not check if references exist as it is too slow"
 	Undeclared declare: self name asSymbol from: Smalltalk globals.
 	myCategory := self category.
 	self environment forgetClass: self logged: logged.
+
+	"In case the class has deprecated aliases we need to remove them from the system dictionary."
+	self deprecatedAliases do: [ :alias | self environment removeKey: alias ifAbsent: [  ] ].
```

Comment:
I'm not sure how strong the name ownership is (or should be). Maybe remove the alias from the environment only if it still refer to the class to be removed?

Reply:
I'm checking that but I see that once we have a global variable installed, if we create a class of the same name, it is not been registered to the system dictionary, 

This seems to big to treat in this specific PR IMO.

Reply:
I'll still do a new commit that manages the case tho